### ISSUE TITLE
Local dotnet format killing PC by running multiple copies

### ISSUE THREAD

**Issue by cvallance** [pre-commit/pre-commit#3069]
### search you tried in the issue tracker

dotnet

### describe your issue

I was setting up a local pre-commit hook for dotnet format as per the second set of instructions here, https://github.com/dotnet/format/blob/main/docs/integrations.md#pre-commitcom-hook-to-reformat 

When running it with a large number of changes it was killing my PC (gobble up my entire 64gb of ram and pegged the CPU).

Running `find . -name "*.cs" -print0 | xargs -0 dotnet format --include` on the directory seemed to run just fine, so I ran an experiment and found that it must be running multiple copies of `dotnet format`.

E.g. if I changed the config to:
```
      - id: test-echo
        name: test-echo
        language: system
        entry: echo BLAH ||
        types_or: [ "c#" ]
        stages: [commit]
        verbose: true
```

And then run `pre-commit run test --all-files` I see `BLAH || files.cs on.cs files.cs on.cs files.cs` 29 times... i.e. it ran the echo command 29 times, which I assume it's doing for dotnet as well.

Is this intended behaviour? Can I change it via some kind of configuration?

### pre-commit --version

pre-commit 3.5.0

### .pre-commit-config.yaml

```yaml
default_install_hook_types: [pre-commit, pre-push]

repos:
  - repo: local
    hooks:
      - id: dotnet-format
        name: dotnet-format
        language: system
        entry: dotnet format --include
        types_or: [ "c#" ]
        stages: [commit]
```

### ~/.cache/pre-commit/pre-commit.log (if present)

NA

**Comment by asottile** [comment]
this repository is for the pre-commit framework -- we don't maintain the dotnet hook -- I'd recommend asking around with someone who wrote that integration.

**Comment by cvallance** [comment]
@asottile Sorry, can you please read the entire issue as it has to do with the behaviour of `pre-commit` that isn't specific to the dotnet format  hook.

**Comment by asottile** [comment]
@cvallance have you read the docs? pre-commit is simply doing what you asked it to do

**Comment by cvallance** [comment]
@asottile Yes I've read the docs. Maybe you can help me by pointing me to the part where it will documents running the `entry` multiple times?

**Comment by asottile** [comment]
I am not a glorified search engine -- please try and read the docs again

**Comment by cvallance** [comment]
Wow, this interaction has been a treat. Thanks.

**Comment by asottile** [comment]
@cvallance put yourself in my shoes -- if every of the millions of pre-commit users posted a github issue without actually reading the docs I'd spend no time actually doing fun things.  actually that's kind of what it's like right now!

**Comment by cvallance** [comment]
Dude, I've read the docs. I'll go back and read them again... but maybe a little assistance in telling me where this particular feature is documented or maybe a term I could use to find it, because as I say, I've already read the docs and I'm likely just to turn back up here again saying I can't find it.

**Comment by cvallance** [comment]
@asottile ... sorry man, I've read [the documentation](https://pre-commit.com/index.html#repository-local-hooks) again and I still can't see anything about how it runs `entry` multiple times for a `Repository local hook`... you're going to have to help me please.

Am I missing something? Is there some other documentation I should be looking at? What should I be searching to find the answer?

**Comment by asottile** [comment]
have you found all the options for configuring a hook and read each of them?

**Comment by cvallance** [comment]
Ok. Found it.

I had to check out the source code, stumble around some python looking for the invocation of entry, and I eventually found `target_concurrency` ... ah, there's `PRE_COMMIT_NO_CONCURRENCY`, maybe I'll search the documentation for "concur" ... huh, zero results. Ok... back to the code, hmmm... `require_serial`, that's better and I can find it in the documentation 🎉 

Thing is, it's under "Creating new hooks" ... which IS NOT what I'm doing. I'm using [Repository local hooks](https://pre-commit.com/index.html#repository-local-hooks) and it mentions nothing about "Creating new hooks" aside from a few required fields.

Anyway, thanks... I guess.
